tests/compiler/inlining.untouched.wat

```diff
@@ -46,6 +46,23 @@
   (local $5 i32)
   (local $6 i32)
   (local $7 i32)
+  (local $8 i32)
+  (local $9 i32)
+  (local $10 i32)
+  (local $11 i32)
+  (local $12 i32)
+  (local $13 i32)
+  (local $14 i32)
+  (local $15 i32)
+  (local $16 i32)
+  (local $17 i32)
+  (local $18 i32)
+  (local $19 i32)
+  (local $20 i32)
+  (local $21 i32)
+  (local $22 i32)
+  (local $23 i32)
+  (local $24 i32)
```

Comment:
Typically looks like this in untouched output, while optimized output only shuffles around some locals, if anything at all.

Reply:
It's interesting that we can't optimize it away 🤔 seems like we should be able to see that the last use of a local occurs before the first use of another local and then make them into the same local

Reply:
Note that this is untouched output without any optimizations. Collapsing these is easy for Binaryen, which does this under any opt level. What I'm interested in here is how much of a problem long local lists actually are, since removing the deduplication we have in the compiler right now would make certain things easier to reason about.